packages/ui/cart/view/CartLineItem.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+   Alert,
+   Box,
+   Collapse,
+   Flex,
+   HStack,
+   IconButton,
+   Link,
+   Text,
+   useToast,
+   VStack,
+} from '@chakra-ui/react';
+import {
+   faCircleExclamation,
+   faCircleMinus,
+   faCirclePlus,
+   faTrash,
+} from '@fortawesome/pro-solid-svg-icons';
+import { useAppContext } from '@ifixit/app';
+import {
+   CartLineItem as LineItem,
+   useRemoveLineItem,
+   useUpdateLineItemQuantity,
+} from '@ifixit/cart-sdk';
+import { multiplyMoney } from '@ifixit/helpers';
+import { FaIcon } from '@ifixit/icons';
+import * as React from 'react';
+import { ProductVariantPrice } from '../../commerce';
+import { CartLineItemImage } from './CartLineItemImage';
+
+interface CartLineItemProps {
+   lineItem: LineItem;
+}
+
+export function CartLineItem({ lineItem }: CartLineItemProps) {
+   const appContext = useAppContext();
+   const toast = useToast();
+   const removeLineItem = useRemoveLineItem();
+   const updateLineItemQuantity = useUpdateLineItemQuantity();
+
+   React.useEffect(() => {
+      if (updateLineItemQuantity.isError) {
+         const id = setTimeout(() => {
+            updateLineItemQuantity.reset();
+         }, 3000);
+         return () => clearTimeout(id);
+      }
+   }, [updateLineItemQuantity.isError, updateLineItemQuantity.reset]);
+
+   React.useEffect(() => {
+      if (removeLineItem.isError) {
+         toast.closeAll();
+         toast({
+            status: 'error',
+            title: 'Unable to remove product',
+            description:
+               'Please try again. If the problem persists contact us.',
+            duration: 5000,
+            isClosable: true,
+            variant: 'subtle',
+            position: 'bottom-right',
+         });
+      }
+   }, [removeLineItem.isError]);
+
+   const incrementQuantity = () => {
+      updateLineItemQuantity.mutate({
+         itemcode: lineItem.itemcode,
+         quantity: 1,
+      });
+   };
+
+   const decrementQuantity = () => {
+      updateLineItemQuantity.mutate({
+         itemcode: lineItem.itemcode,
+         quantity: -1,
+      });
+   };
+
+   const handleRemoveLineItem = () => {
+      removeLineItem.mutate({
+         itemcode: lineItem.itemcode,
+      });
+   };
+
+   return (
+      <HStack
+         w="full"
+         justify="space-between"
+         p="3"
+         spacing="3"
+         align="stretch"
+      >
+         <CartLineItemImage src={lineItem.imageSrc} alt={lineItem.name} />
+         <VStack align="flex-start" spacing="0" flexGrow="1">
+            <Link
+               href={`${appContext.ifixitOrigin}/Store/Product/${lineItem.itemcode}`}
```

Comment:
What is this link? Should this be /products/{handle}? Or is there a url property on the lineItem?

Reply:
Good catch. I'm not sure, though it's the same as the current cart-drawer:

https://github.com/iFixit/react-commerce/blob/7726466185b364acee4b92f2e7a9124ade6e724d/packages/ui/cart/drawer/CartLineItem.tsx#L95

Looks like a redirecter: https://www.ifixit.com/Store/Product/IF145-307-4

Though... I can't yet figure out where it's from.

Reply:
Definitely php code because I see varnish headers in the response.

Reply:
I logged the route and got here:
![image](https://github.com/iFixit/react-commerce/assets/26855/576294ce-dede-445d-936f-56bb58c23447)

Which ends up here for the redirect:
https://github.com/iFixit/ifixit/blob/72d86ea35f0b850df0848070df8eb5980f613d00/Cart/UI/parts_store.php#L120

Crazy.

Reply:
I think this was here before we had the product pages in next.js and has never been updated. It think we can safely refactor it now 👍